src/sources/kafka.rs

```diff
@@ -160,6 +162,10 @@ fn create_consumer(config: KafkaSourceConfig) -> crate::Result<StreamConsumer> {
             &config.commit_interval_ms.to_string(),
         )
         .set("enable.auto.offset.store", "false")
+        .set(
+            "compression.codec",
```

Comment:
This is only a producer-side configuration item. The compression format of a record batch is encoded at write time and the consumer will check the header and decompress accordingly.

Reply:
Hmm, ok, the original issue asked for support in both the sink and source. Reading [this Kafka document](https://cwiki.apache.org/confluence/display/KAFKA/Compression) clarifies that this indeed is a producer-only. Is the `compression.topics` configuration setting also required to make this work?

Reply:
I don't think we need to be concerned with the topic-level configuration option. That's more for management of the kafka cluster itself.

I believe the `Not implemented` error we saw was just due to not having the relevant features enabled on the crate, so it did not have an available implementation of the decompression algorithm for that record batch. There is not setting to tell a consumer how to decompress a batch because that information is encoded as part of the wire protocol.